### PR TITLE
Use whitelist to restrict urlReplacement for scoped analytics element

### PR DESCRIPTION
**Pull request by zhouyx**
@dvoytenko  As we agreed, used whitelist to apply the restriction as the first step
I created #8396 to for future review of adding extra variables to the whitelist.

### CODE REVIEW

File: extensions/amp-analytics/0.1/url-replacement-whitelists.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+ * limitations under the License.
+ */
+
+/**
+ * Used for inserted scoped analytics element.
+ * @const {!Object<string, boolean>}
+ */
+export const ScopedElementWhiteList = {
+  // TODO: Please Review the list!
+  'RANDOM': true,
```
Comment (lannka) [inline extensions/amp-analytics/0.1/url-replacement-whitelists.js]: I suggest we turn off everything for now so we can get the PR in independently of a privacy review.

Comment (lannka) [inline extensions/amp-analytics/0.1/url-replacement-whitelists.js]: i suggest we only keep the `true`s to shorten the list.

Comment (zhouyx) [inline extensions/amp-analytics/0.1/url-replacement-whitelists.js]: That's what I plan to do eventually. I put the complete list for the convenience of privacy review. But If we turn off everything for now, I can definitely do this now.

Comment (zhouyx) [inline extensions/amp-analytics/0.1/url-replacement-whitelists.js]: even for variables like `RAMDOM` ?

File: extensions/amp-analytics/0.1/amp-analytics.js [hunk 2]
```diff
@@ -77,6 +78,9 @@ export class AmpAnalytics extends AMP.BaseElement {
      */
     this.type_ = null;
 
+    /** @private {!boolean} */
+    this.isScoped_ = !!element.getAttribute('scope');
```
Comment (dvoytenko) [inline extensions/amp-analytics/0.1/amp-analytics.js]: `element.hasAttribute`

File: extensions/amp-analytics/0.1/url-replacement-whitelists.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+  'COUNTER': true,
+  'CANONICAL_URL': false,
```
Comment (dvoytenko) [inline extensions/amp-analytics/0.1/url-replacement-whitelists.js]: CANONICAL should be allowed.

File: extensions/amp-analytics/0.1/url-replacement-whitelists.js [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+  'CANONICAL_HOST': false,
+  'CANONICAL_HOSTNAME': false,
+  'CANONICAL_PATH': false,
+  'DOCUMENT_REFERRER': false,
+  'TITLE': true,
+  'AMPDOC_URL': false,
```
Comment (dvoytenko) [inline extensions/amp-analytics/0.1/url-replacement-whitelists.js]: AMPDOC_ URL stuff is good too.

File: extensions/amp-analytics/0.1/url-replacement-whitelists.js [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+  'AMPDOC_HOST': false,
+  'AMPDOC_HOSTNAME': false,
+  'SOURCE_URL': false,
```
Comment (dvoytenko) [inline extensions/amp-analytics/0.1/url-replacement-whitelists.js]: SOURCE_ as well

File: extensions/amp-analytics/0.1/amp-analytics.js (hunk 2 above)
Comment (zhouyx) [inline extensions/amp-analytics/0.1/amp-analytics.js]: done

File: extensions/amp-analytics/0.1/url-replacement-whitelists.js (hunk 3 above)
Comment (zhouyx) [inline extensions/amp-analytics/0.1/url-replacement-whitelists.js]: I assume all `CANONICAL_URL`, `CANONICAL_HOST`, `CANONICAL_HOSTNAME`, `CANONICAL_PATH` are good. I set all of them to true.

**Comment by zhouyx** [comment]
@lannka convinced me that we should first turn off everything in the whitelist. And only turn them on if anyone request.

**Comment by dvoytenko** [comment]
@zhouyx I think it's safe to turn on AMPDOC_* vars and if we do that, it's the same as enabling CANONICAL_ and SOURCE_. 3p ampcontext is a good source for this data and canonical, et all are allowed there. So, let's enable these right away.

**Comment by dvoytenko** [comment]
Otherwise, LGTM

File: extensions/amp-analytics/0.1/url-replacement-whitelists.js (hunk 1 above)
Comment (zhouyx) [inline extensions/amp-analytics/0.1/url-replacement-whitelists.js]: fixed.

File: extensions/amp-analytics/0.1/amp-analytics.js [hunk 6]
```diff
@@ -77,6 +78,9 @@ export class AmpAnalytics extends AMP.BaseElement {
      */
     this.type_ = null;
 
+    /** @private {!boolean} */
+    this.isScoped_ = element.hasAttribute('scope');
```
Comment (lannka) [inline extensions/amp-analytics/0.1/amp-analytics.js]: shall we give it a different name? `scope` is already used in `selectionMethod` which has a quite different meaning.
how about `<amp-analytics sandbox>`

File: extensions/amp-analytics/0.1/url-replacement-whitelists.js [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Used for inserted scoped analytics element.
+ * @const {!Object<string, boolean>}
+ */
+export const ScopedElementWhiteList = {
```
Comment (lannka) [inline extensions/amp-analytics/0.1/url-replacement-whitelists.js]: SANDBOX_AVAILABLE_VARS

maybe also change the file name.

**Pull request merged**